jupyterlab_gitsync/src/service/notebook_file.ts

```diff
@@ -41,55 +50,100 @@ export class NotebookFile implements IFile {
     return this._dirtyState;
   }
 
+  get activeCell() {
+    return this.content.activeCell;
+  }
+
+  get activeCellIndex(): number {
+    return this.content.activeCellIndex;
+  }
+
   async save() {
     try{
-      await this.context.save();
+      const content = (this.content.model as NotebookModel).toJSON();
+      await this._saveFile(content);
+      this.resolver.addVersion(content, 'base');
     } catch (error) {
       console.warn(error);
     }
   }
 
   async reload() {
+    await this._getRemoteVersion();
     this._getLocalVersion();
-    this._getRemoteVersion();
+    this._getEditorView();
+    const merged = await this.resolver.mergeVersions();
+    if (merged) { await this._displayText(merged); }
+  }
+
+  private async _displayText(merged) {
+    await this._saveFile(merged)
     await this.context.revert();
```

Comment:
what does this context revert do?

Reply:
context revert displays the disk contents to the user. This is necessary so that when we save, we do not get a pop-up stating that the disk contents have changed since the last time we opened the file (something to do with the last modified time).